src/git/sync_common.ml

```diff
@@ -0,0 +1,147 @@
+module Make
```

Comment:
can you move this into the `Sync` module? Not sure we need/want to create a new module name for this (people will wonder what's the difference between `Sync` and `Sync_common`)